### PR TITLE
use assertj methods

### PR DESCRIPTION
**Pull request by Bananeweizen**
Use more specialized assertj expectations than just .isTrue().

### CODE REVIEW

**Comment by Lineflyer** [comment]
retest this please

File: tests/src-android/cgeo/geocaching/models/GeocacheTest.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public static void testMergeLocalUserModifiedCoordsAndServerSideModified() {
 
         assertThat(download.hasUserModifiedCoords()).as("merged user modified").isEqualTo(true);
         assertThat(download.getCoords()).as("merged coordinates").isEqualTo(new Geopoint(41.0, 9.0));
-        assertThat(download.getWaypoints().size()).as("merged waypoints size").isEqualTo(1);
+        assertThat(download.getWaypoints()).as("merged waypoints size").hasSize(1);
```
Comment (UniQP) [inline tests/src-android/cgeo/geocaching/models/GeocacheTest.java]:
```suggestion
        assertThat(download.getWaypoints()).as("merged waypoints").hasSize(1);
```

File: tests/src-android/cgeo/geocaching/filter/TerrainFilterTest.java [hunk 2]
```diff
@@ -22,6 +22,6 @@ public static void testTerrainFilter() {
     }
 
     public static void testAllFilters() {
-        assertThat(new TerrainFilter.Factory().getFilters().size() == 7); // terrain ranges from 1 to 7 (due to ExtremCaching.com using that value).isTrue()
+        assertThat(new TerrainFilter.Factory().getFilters()).hasSize(7); // terrain ranges from 1 to 7 (due to ExtremCaching.com using that value).isTrue()
```
Comment (UniQP) [inline tests/src-android/cgeo/geocaching/filter/TerrainFilterTest.java]:
```suggestion
        assertThat(new TerrainFilter.Factory().getFilters()).hasSize(7); // terrain ranges from 1 to 7 (due to ExtremCaching.com using that value)
```

File: tests/src/cgeo/geocaching/network/ParametersTest.java [hunk 3]
```diff
@@ -107,7 +107,7 @@ public void testMerge() {
         assertThat(Parameters.merge(params1, null)).isSameAs(params1);
         assertThat(Parameters.merge(null, params1)).isSameAs(params1);
         assertThat(Parameters.merge(null, params1, null, params2, null)).isSameAs(params1);
-        assertThat(params1.size() == 4);
-        assertThat(params2.size() == 2);
+        assertThat(params1).hasSize(4);
```
Comment (UniQP) [inline tests/src/cgeo/geocaching/network/ParametersTest.java]:
```suggestion
        assertThat(params1).hasSize(2);
```

File: tests/src/cgeo/geocaching/network/ParametersTest.java [hunk 4]
```diff
@@ -107,7 +107,7 @@ public void testMerge() {
...
+        assertThat(params2).hasSize(2);
```
Comment (UniQP) [inline tests/src/cgeo/geocaching/network/ParametersTest.java]:
```suggestion
        assertThat(params2).hasSize(1);
```

Comment (UniQP) [reply tests/src/cgeo/geocaching/network/ParametersTest.java]: The previous code missed the `isTrue()` calls and had wrong expectations.

**Pull request merged**